src/pages/Compare/index.spec.tsx

```diff
@@ -1361,4 +1380,133 @@ describe(__filename, () => {
   });
+
+  describe('updateSelectedPath', () => {
+    const setUpForPathUpdateAndRender = ({
+      baseVersionId,
+      headVersionId,
+      initialPath,
+      lastSelectedPath,
+      location,
+      pathList,
+    }: {
+      baseVersionId: number;
+      headVersionId: number;
+      initialPath: string;
+      lastSelectedPath: string;
+      location: Location;
+      pathList: string[];
+    }) => {
+      const history = createFakeHistory({ location });
+      const version = createExternalVersionWithEntries(
+        pathList.map((path) => ({ path })),
+        { id: headVersionId, selected_file: initialPath },
+      );
+
+      return loadDiffAndRender({
+        baseVersionId,
+        headVersionId,
+        history,
+        lastSelectedPath,
+        version,
+      });
+    };
+
+    it('updates the selected path if it is different from the path in the URL', () => {
+      const baseVersionId = nextUniqueId();
+      const headVersionId = baseVersionId + 1;
+      const path1 = 'a/path';
+      const path2 = 'b/path';
+      const location = createFakeLocation({
+        search: queryString.stringify({ path: path2 }),
+      });
+
+      const { dispatchSpy } = setUpForPathUpdateAndRender({
+        baseVersionId,
+        headVersionId,
+        initialPath: path1,
+        lastSelectedPath: path1,
+        location,
+        pathList: [path1, path2],
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        versionsActions.updateSelectedPath({
+          selectedPath: path2,
+          versionId: headVersionId,
+        }),
+      );
+    });
+
+    it('updates the selected path to the initial value if the path is undefined in the URL', () => {
+      const baseVersionId = nextUniqueId();
+      const headVersionId = baseVersionId + 1;
+      const initialPath = 'a/path';
+      const lastSelectedPath = 'b/path';
+
+      const { dispatchSpy } = setUpForPathUpdateAndRender({
+        baseVersionId,
+        headVersionId,
+        initialPath,
+        lastSelectedPath,
+        location: createFakeLocation({ search: '' }),
+        pathList: [initialPath, lastSelectedPath],
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(
+        versionsActions.updateSelectedPath({
+          selectedPath: initialPath,
+          versionId: headVersionId,
+        }),
+      );
+    });
+
+    it('does not update the selected path if it has been updated', () => {
+      const baseVersionId = nextUniqueId();
+      const headVersionId = baseVersionId + 1;
+      const path = 'a/path';
+      const location = createFakeLocation({
+        search: queryString.stringify({ path }),
+      });
+
+      const { dispatchSpy } = setUpForPathUpdateAndRender({
+        baseVersionId,
+        headVersionId,
+        initialPath: path,
+        lastSelectedPath: path,
+        location,
+        pathList: [path],
+      });
+
+      expect(dispatchSpy).not.toHaveBeenCalledWith(
+        versionsActions.updateSelectedPath({
```

Comment:
I'm a bit concerned that we're being so specific in this assertion. I.e., we're testing that 
`versionsActions.updateSelectedPath` wasn't dispatched with `selectedPath: path` and `versionId: headVersionId`, but really we want to assert that `versionsActions.updateSelectedPath` wasn't dispatched at all, regardless of what the `path` was.

One way to do this would be to change this to something like:

```js
expect(dispatchSpy).not.toHaveBeenCalledWith(
	versionsActions.updateSelectedPath(
		expect.objectContaining({
          versionId: headVersionId,
        })
	)
)
```

or, if it's safe to assert that `dispatch` wasn't called at all (which might not be the case), we could simply do:

`expect(dispatchSpy).not.toHaveBeenCalled()`

Reply:
I went with the first way (use `expect.objectContaining`), since dispatch was called here.